File: app/src/main/java/com/mtp/laboproject/view/ui/fragment/AlertFragment.java
package com.mtp.laboproject.view.ui.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.mtp.laboproject.R;
import com.mtp.laboproject.data.model.alert.AlertsDetailsResponse;
import com.mtp.laboproject.data.model.alert.AlertsResponse;
import com.mtp.laboproject.view.adapter.AlertsAdapter;
import com.mtp.laboproject.view.listener.AlertsClickListener;
import com.mtp.laboproject.view.viewmodel.AlertViewModel;

import org.jetbrains.annotations.NotNull;
import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

public class AlertFragment extends BaseFragment implements AlertsClickListener {
    private AlertViewModel mViewModel;
    private AlertsAdapter alertsAdapter;

    @BindView(R.id.recycleview_alert)
    public RecyclerView recycleview_alert;


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v= inflater.inflate(R.layout.fragment_alert, container, false);
        //ButterKnife.inject(this,v);
        recycleview_alert=v.findViewById(R.id.recycleview_alert);
        ButterKnife.bind(this,v);

        return v;

    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        setupAlertList();

    }

    private void setupAlertList() {
        mViewModel = ViewModelProviders.of(this).get(AlertViewModel.class);
        mViewModel.getAlerts();
        mViewModel.alertsLiveData.observe(getViewLifecycleOwner(), (Observer<AlertsResponse>) value -> {
            if (value != null) {
                LinearLayoutManager layoutManager = new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false);
                recycleview_alert.setLayoutManager(layoutManager);
                alertsAdapter= new AlertsAdapter(new ArrayList(value.getData()),this);
                recycleview_alert.setAdapter(alertsAdapter);
            }
        });
    }

    @Override
    public void onRecyclerViewItemClick(@NotNull View view, @NotNull AlertsDetailsResponse labo) {

    }
}

File: app/src/main/java/com/mtp/laboproject/view/ui/fragment/AlertFragment.kt
package com.mtp.laboproject.view.ui.fragment

import androidx.lifecycle.ViewModelProviders
import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.GridLayoutManager
import com.mtp.laboproject.R
import com.mtp.laboproject.data.model.LaboratoryListResponse
import com.mtp.laboproject.view.adapter.LaboratoryAdapter
import com.mtp.laboproject.listener.LaboratoryClickListener
import com.mtp.laboproject.view.ui.DetailsLaboBottomSheet
import com.mtp.laboproject.view.viewmodel.LabsViewModel
import com.mtp.laboproject.view.factory.LabsViewModelFactory
import kotlinx.android.synthetic.main.fragment_laboratory.*
import android.view.MenuInflater
import android.widget.LinearLayout
import android.widget.SearchView
import androidx.recyclerview.widget.LinearLayoutManager
import com.mtp.laboproject.data.model.AlertsDetailsResponse
import com.mtp.laboproject.listener.AlertsClickListener
import com.mtp.laboproject.view.adapter.AlertsAdapter
import com.mtp.laboproject.view.factory.AlertsViewModelFactory
import com.mtp.laboproject.view.viewmodel.AlertsViewModel
import kotlinx.android.synthetic.main.fragment_alert.*


class AlertFragment : BaseFragment(), AlertsClickListener {

    private lateinit var alertsViewModel: AlertsViewModel
    private lateinit var alertsAdapter: AlertsAdapter

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_alert, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        //setAlert()
    }

    private fun setAlert() {
        val factory = AlertsViewModelFactory()
        alertsViewModel = ViewModelProviders.of(this, factory).get(AlertsViewModel::class.java)
        alertsViewModel.getAlerts()
        alertsViewModel.alertsLiveData.observe(viewLifecycleOwner, Observer { alert ->
            recycleview_alerts.also {
                it.layoutManager = LinearLayoutManager(context)
                it.setHasFixedSize(true)
               // alertsAdapter = AlertsAdapter(alert, this)
               // it.adapter = alertsAdapter
            }
        })
    }

    override fun onRecyclerViewItemClick(view: View, labo: AlertsDetailsResponse) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

}

